libraries/chain/include/eosio/chain/transaction_metadata.hpp

```diff
@@ -50,13 +50,14 @@ class transaction_metadata {
          signed_id = digest_type::hash(*packed_trx);
       }
 
-      const flat_set<public_key_type>& recover_keys( const chain_id_type& chain_id );
-
       // must be called from main application thread
-      // signing_keys_future should only be accessed by main application thread
       static signing_keys_future_type
-      create_signing_keys_future( const transaction_metadata_ptr& mtrx, boost::asio::thread_pool& thread_pool,
-                                  const chain_id_type& chain_id, fc::microseconds time_limit );
+      start_recover_keys( const transaction_metadata_ptr& mtrx, boost::asio::thread_pool& thread_pool,
```

Comment:
I think this should be a member function rather than a static function. And it should not return anything. The `signing_keys_future` can be made protected.

Reply:
Can't be a member function unless we derive transaction_metdata from `enable_shared_from_this`. I'm ok with deriving from `enabled_shared_from_this` but that is the reason it is static.

Reply:
No, it's okay. We can just stick with minimal changes for this patch release.